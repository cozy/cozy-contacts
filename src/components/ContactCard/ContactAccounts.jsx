import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'cozy-ui/transpiled/react/I18n'
import Typography from 'cozy-ui/transpiled/react/Typography'
import Icon from 'cozy-ui/transpiled/react/Icon'

import FieldsetTitle from '../Common/FieldsetTitle'
import Fieldset from '../Common/Fieldset'

const ACCOUNTS_MAPPING = {
  'konnector-google': {
    label: 'Google',
    icon: 'google'
  }
}

const ContactAccounts = ({ accounts, t }) => (
  <>
    <FieldsetTitle
      title={t('associated_accounts', { smart_count: accounts.length })}
    />
    <Fieldset>
      {accounts.map(account => {
        const { icon, label } = ACCOUNTS_MAPPING[account.type]
        return (
          <li key={account.id}>
            <div className="u-flex u-mt-1 u-mb-half">
              <div className="u-mr-1">
                <Icon icon={icon} />
              </div>
              <div>
                <div className="u-mb-half u-breakword">
                  <Typography variant="body1">{label}</Typography>
                  <Typography variant="caption" color="textSecondary">
                    {account.name}
                  </Typography>
                </div>
              </div>
            </div>
          </li>
        )
      })}
    </Fieldset>
  </>
)

ContactAccounts.propTypes = {
  accounts: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      type: PropTypes.string.isRequired
    })
  )
}

export default translate()(ContactAccounts)
