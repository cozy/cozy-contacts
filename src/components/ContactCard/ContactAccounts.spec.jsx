import React from 'react'
import renderer from 'react-test-renderer'
import ContactAccounts from './ContactAccounts'
import AppLike from '../../tests/Applike'

describe('ContactAccounts component', () => {
  it('should render a list of contact accounts', () => {
    const props = {
      accounts: [
        {
          id: '505ec401-1302-4073-b41d-aee2d0f4dcbd',
          name: 'john.doe@gmail.com',
          type: 'konnector-google'
        }
      ]
    }
    const tree = renderer
      .create(
        <AppLike>
          <ContactAccounts {...props} />
        </AppLike>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
