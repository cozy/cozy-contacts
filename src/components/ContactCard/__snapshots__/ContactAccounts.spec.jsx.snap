// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactAccounts component should render a list of contact accounts 1`] = `
Array [
  <h3
    className="u-title-h2 u-mt-1-half u-mb-1"
  >
    Account linked to this contact
  </h3>,
  <ol
    className="u-nolist u-m-0 u-p-0 u-pl-1-half u-pl-0-s"
  >
    <li>
      <div
        className="u-flex u-mt-1 u-mb-half"
      >
        <div
          className="u-mr-1"
        >
          <svg
            className="styles__icon___23x3R"
            height="16"
            style={Object {}}
            width="16"
          >
            <use
              xlinkHref="#google"
            />
          </svg>
        </div>
        <div>
          <div
            className="u-mb-half u-breakword"
          >
            <p
              className="MuiTypography-root-1 MuiTypography-body1-10"
            >
              Google
            </p>
            <span
              className="MuiTypography-root-1 MuiTypography-caption-11 MuiTypography-colorTextSecondary-34"
            >
              john.doe@gmail.com
            </span>
          </div>
        </div>
      </div>
    </li>
  </ol>,
]
`;
